sdk/typescript/src/cryptography/multisig.ts

```diff
@@ -57,20 +58,41 @@ export function toMultiSigAddress(pks: PubkeyWeightPair[], threshold: number): s
 	if (pks.length > MAX_SIGNER_IN_MULTISIG) {
 		throw new Error(`Max number of signers in a multisig is ${MAX_SIGNER_IN_MULTISIG}`);
 	}
+
+	if (pks.length < MIN_SIGNER_IN_MULTISIG) {
+		throw new Error(`Min number of signers in a multisig is ${MIN_SIGNER_IN_MULTISIG}`);
+	}
+
 	// max length = 1 flag byte + (max pk size + max weight size (u8)) * max signer size + 2 threshold bytes (u16)
 	let maxLength = 1 + (64 + 1) * MAX_SIGNER_IN_MULTISIG + 2;
 	let tmp = new Uint8Array(maxLength);
 	tmp.set([SIGNATURE_SCHEME_TO_FLAG['MultiSig']]);
 
 	let arr = to_uint8array(threshold);
 	tmp.set(arr, 1);
+	// The initial value 3 ensures that following data will be after the flag byte and threshold bytes
 	let i = 3;
+	let totalWeight = 0;
+	const seenPubKeys = new Set<PublicKey>();
+
 	for (const pk of pks) {
+		if (pk.weight < 1 || pk.weight > 255) {
+			throw new Error(`Invalid weight`);
+		}
+		if (seenPubKeys.has(pk.pubKey)) {
```

Comment:
@joyqvq 
this logic deviates from our rust implementation, and I believe it is an improvement we should carry over there.